audformat/core/table.py

```diff
@@ -1479,6 +1479,47 @@ def get(
 
         return result
 
+    def map_files(
+            self,
+            func: typing.Callable[[str], str],
+    ):
+        r"""Apply function to file names in table.
+
+        Relies on :meth:`pandas.Index.map`,
+        which can be slow.
+        If speed is crucial,
+        consider to change the index directly.
+        In the following example we prefix every file with a folder:
+
+        .. code-block:: python
+
+            root = '/root/'
+            if table.is_filewise:
+                table.df.index = root + table.df.index
+                table.df.index.name = audformat.define.IndexField.FILE
+            elif len(table.df.index) > 0:
+                table.df.index.set_levels(
+                    root + table.df.index.levels[0],
+                    audformat.define.IndexField.FILE,
+                    inplace=True,
+                )
+
+        Args:
+            func: map function
+
+        """
+        if self.is_segmented:
+            index = self.df.index.map(
+                lambda x: (func(x[0]), x[1], x[2])
+            )
+            index = utils.set_index_dtypes(
+                index,
+                {define.IndexField.FILE: 'string'},
+            )
+            self.df.index = index
+        else:
+            self.df.index = self.df.index.map(lambda x: func(x))
```

Comment:
I think we should use
```python
self.df.index = audformat.utils.map_file_path(table.df.index, func)
```
here and make sure that the `string` dtype is preserved inside that function.

Reply:
Changed it accordingly. It's a bit unfortunate we have chosen a different name for the utility function.